### DISCUSSION TITLE
Dev environment setup help

### DISCUSSION THREAD

**Discussion by Kanary159357** [Budibase/budibase#6696] [General]
**Hosting**
I want to contribute to budibase so 
I `git clone` budibase repogitory and then 
do `yarn setup`.

The problem is after installing, when I access to localhost:4001, It has error 

https://user-images.githubusercontent.com/69702813/178883419-2fde3b9b-42f4-4719-b53a-4fbe181c92a8.mov

Yarn Version : 1.22.17

OS: MacBook 16inch Montrey M1

Console error message is here

`[audit] [tenant=83cecc2d72be4d1fa9187190ebd6607e_default] [identityType=user] [identity=us_1d9aa0afa3284947a3333c0f193e2896]  auth:login 
@budibase/worker: ERROR [1657764811222] (9080 on Kanary-EX-MAC.local):
@budibase/worker:     req: {
@budibase/worker:       "id": 9,
@budibase/worker:       "method": "GET",
@budibase/worker:       "url": "/api/global/self?",
@budibase/worker:       "headers": {
@budibase/worker:         "host": "localhost:4001",
@budibase/worker:         "connection": "keep-alive",
@budibase/worker:         "sec-ch-ua": "\".Not/A)Brand\";v=\"99\", \"Google Chrome\";v=\"103\", \"Chromium\";v=\"103\"",
@budibase/worker:         "accept": "application/json",
@budibase/worker:         "x-budibase-app-id": "",
@budibase/worker:         "x-budibase-api-version": "1",
@budibase/worker:         "sec-ch-ua-mobile": "?0",
@budibase/worker:         "user-agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36",
@budibase/worker:         "sec-ch-ua-platform": "\"macOS\"",
@budibase/worker:         "sec-fetch-site": "same-origin",
@budibase/worker:         "sec-fetch-mode": "cors",
@budibase/worker:         "sec-fetch-dest": "empty",
@budibase/worker:         "referer": "http://localhost:4001/builder/auth/login",
@budibase/worker:         "accept-encoding": "gzip, deflate, br",
@budibase/worker:         "accept-language": "ko-KR,ko;q=0.9,en-US;q=0.8,en;q=0.7",
@budibase/worker:         "cookie": "_ga=GA1.1.251928055.1649317694; budibase:init=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE2NDkzODg3ODR9.FcKlTojO6ElOFjYxw1KFO7Upu3EUficSaCiVdaWiuJg; budibase:init.sig=6DljCSUMnhf9yQkV_Rfb6CerVuc; budibase:auth.sig=sqR86ozl1jZaWq-WbDH4SkO8Ox4; ch-veil-id=79b8c82c-4c47-493b-bd83-658900185840; _ga_YP079TYK6L=GS1.1.1657688844.16.1.1657691975.0; _ga_CONFIG=GS1.1.1657688844.13.1.1657691975.0; budibase:currentapp=; budibase:returnurl=/builder"
@budibase/worker:       },
@budibase/worker:       "remoteAddress": "::1",
@budibase/worker:       "remotePort": 58067
@budibase/worker:     }
@budibase/worker:     err: {
@budibase/worker:       "type": "TypeError",
@budibase/worker:       "message": "Cannot convert undefined or null to object",
@budibase/worker:       "stack":
@budibase/worker:           TypeError: Cannot convert undefined or null to object
@budibase/worker:               at /Users/kanary/budibase/packages/backend-core/src/middleware/authenticated.js:131:9
@budibase/worker:               at Generator.next (<anonymous>)
@budibase/worker:               at /Users/kanary/budibase/packages/backend-core/src/middleware/authenticated.js:8:71
@budibase/worker:               at new Promise (<anonymous>)
@budibase/worker:               at __awaiter (/Users/kanary/budibase/packages/backend-core/src/middleware/authenticated.js:4:12)
@budibase/worker:               at /Users/kanary/budibase/packages/backend-core/src/middleware/authenticated.js:62:30
@budibase/worker:               at dispatch (/Users/kanary/budibase/packages/worker/node_modules/koa-compose/index.js:42:32)
@budibase/worker:               at /Users/kanary/budibase/packages/worker/node_modules/@koa/router/lib/router.js:368:16
@budibase/worker:               at dispatch (/Users/kanary/budibase/packages/worker/node_modules/koa-compose/index.js:42:32)
@budibase/worker:               at /Users/kanary/budibase/packages/worker/node_modules/koa-compress/lib/index.js:38:11
@budibase/worker:               at dispatch (/Users/kanary/budibase/packages/worker/node_modules/koa-compose/index.js:42:32)
@budibase/worker:               at /Users/kanary/budibase/packages/worker/node_modules/@koa/router/lib/router.js:368:16
@budibase/worker:               at dispatch (/Users/kanary/budibase/packages/worker/node_modules/koa-compose/index.js:42:32)
@budibase/worker:               at /Users/kanary/budibase/packages/worker/node_modules/koa-compose/index.js:34:12
@budibase/worker:               at dispatch (/Users/kanary/budibase/packages/worker/node_modules/@koa/router/lib/router.js:373:31)
@budibase/worker:               at dispatch (/Users/kanary/budibase/packages/worker/node_modules/koa-compose/index.js:42:32)
@budibase/worker:       "expose": true,
@budibase/worker:       "statusCode": 403,
@budibase/worker:       "status": 403
@budibase/worker:     }
@budibase/server: ERROR [1657764811223] (9089 on Kanary-EX-MAC.local): Cannot convert undefined or null to object
@budibase/server:     BadRequestError: Cannot convert undefined or null to object
@budibase/server:         at Object.throw (/Users/kanary/budibase/packages/server/node_modules/koa/lib/context.js:97:11)
@budibase/server:         at /Users/kanary/budibase/packages/server/src/api/controllers/dev.js:30:14
@budibase/server:         at Generator.next (<anonymous>)
@budibase/server:         at fulfilled (/Users/kanary/budibase/packages/server/src/api/controllers/dev.js:5:58)
@budibase/server:         at processTicksAndRejections (node:internal/process/task_queues:96:5)
@budibase/server: Trace: BadRequestError: Cannot convert undefined or null to object
@budibase/server:     at Object.throw (/Users/kanary/budibase/packages/server/node_modules/koa/lib/context.js:97:11)
@budibase/server:     at /Users/kanary/budibase/packages/server/src/api/controllers/dev.js:30:14
@budibase/server:     at Generator.next (<anonymous>)
@budibase/server:     at fulfilled (/Users/kanary/budibase/packages/server/src/api/controllers/dev.js:5:58)
@budibase/server:     at processTicksAndRejections (node:internal/process/task_queues:96:5)
@budibase/server:     at /Users/kanary/budibase/packages/server/src/api/index.js:77:15
@budibase/server:     at Generator.throw (<anonymous>)
@budibase/server:     at rejected (/Users/kanary/budibase/packages/server/src/api/index.js:6:65)
@budibase/server:     at processTicksAndRejections (node:internal/process/task_queues:96:5)`

**To Reproduce**
1. git clone 2022-07-14's current budibase repogitory
2. use yarn setup
3. access to `localhost:4001`
4. login and toast popped up
5. but Error

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

**Additional context**

**Comment by aptkingston** [comment]
Hi @Kanary159357. Thanks for wanting to help out! When you run the `setup` command, does everything look alright in the terminal logs? All the docker containers are starting correctly etc? It might take a while on first run for the various images to be pulled. 

Could you also try accessing the builder on port 10000 rather than 4001 - 10000 is the port for our proxy which routes everything correctly and that's how you should access the platform.

Also going to convert this to a discussion for now since this is just support for getting a dev environment setup rather than a bug.

**Reply by Kanary159357** [reply]
Thx accessing to localhost:10000 works well
